src/tests/Microsoft.PowerFx.Interpreter.Tests/DefeeredTypeTests.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Types;
+using Xunit;
+
+namespace Microsoft.PowerFx.Interpreter
+{
+    public class DefeeredTypeTests
+    {
+        [Theory]
+        [InlineData("X")]
+        [InlineData("$\"Test {X}\"")]
+        [InlineData("!X")]
+        [InlineData("Table(X)")]
+        [InlineData("X.Field1")]
+        [InlineData("X + 1")]
+        [InlineData("X + Date(2022, 10, 27)")]
+        [InlineData("X * 1")]
+        [InlineData("X < 2")]
+        [InlineData("\"The\" in X")]
+        [InlineData("123 in X")]
+        [InlineData("123 in Table(X)")]
+        [InlineData("IsBlank(X)")]
+        [InlineData("IsError(X)")]
+        [InlineData("Text(X)")]
+        [InlineData("Value(X)")]
+        [InlineData("Boolean(X)")]
+        [InlineData("Index([1,2,3].Value, X)")]
+
+        // Ensures expression binds without any errors - but issues a warning for the deferred(unknown) type.
+        public void DeferredTypeTest_EnableDeferredType(string script)
+        {
+            Preview.FeatureFlags.StringInterpolation = true;
+            var symbolTable = new SymbolTable();
+            symbolTable.AddVariable("X", FormulaType.Unknown);
+            TestDeferredTypeBindingWarning(script, Features.EnableDeferredType, symbolTable);
+        }
+
+        [Theory]
+        [InlineData("$\"Test {X} {R}\"", "Invalid argument type (Record). Expecting a Text value instead.")]
+        [InlineData("Table(X, N)", "Cannot use a non-record value in this context")]
+        [InlineData("X.field + N.field", "Invalid use of '.'")]
+        [InlineData("Index([1,2,3], X).missing", "Name isn't valid. 'missing' isn't recognized")]
```

Comment:
also, `With(x, ...)`, `ForAll(x, ...)`

Reply:
In current iteration, we are not allowing deferred type on functions which takes formula as an input in any arg.